### PR TITLE
PREQ-306

### PR DESCRIPTION
**Pull request by bubskee**

### CODE REVIEW

File: src/hal/token.clj [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+(defn- kube-secret
+  "access the kubernetes secret"
+  []
+  (slurp (io/file "/etc/secrets/webtoken/private.key")))
+
+(def ^:private secret
+  (if (kubernetes?)
+    (kube-secret)
+    ;;sufficiently obfuscated?
+    (String. (byte-array '(75 118 67 109 97 115 88 122 82 77 67 84 104 73 107)))))
```
Comment (Scriptor) [inline src/hal/token.clj]: NSA got nothin' on us.

File: test/hal/core_test.clj [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    (is (byte-array? raw-key))
+    (is (not (empty? raw-key))))
+  (is (thrown? Exception (get-raw-key :invalid))))
+
+(deftest get-cipher-test
+  (let [iv-bytes (get-iv-bytes)]
+    (let [cipher (get-cipher Cipher/ENCRYPT_MODE "secret" iv-bytes)]
+      (is (instance? Cipher cipher))
+      (is (= (seq iv-bytes) (seq (.getIV cipher)))))
+    (is (thrown? Exception (get-cipher "mode" "secret" iv-bytes)))
```
Comment (Scriptor) [inline test/hal/core_test.clj]: Can we get any more specific than Exception?

File: src/hal/core.clj [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+(def byte-array-class (Class/forName "[B"))
+
+(defn byte-array?
+  "Tests if a value is a byte[]. Returns true/false
+  object - the object to test"
+  [object]
+  (= byte-array-class (type object)))
+
+(defn get-iv-bytes
```
Comment (Scriptor) [inline src/hal/core.clj]: Something here should explain what iv stands for.

File: src/hal/core.clj [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    (.nextBytes sr iv-bytes)
+    iv-bytes))
+
+(defn get-raw-key
+  "Returns an AES SecretKey encoded as a byte[]
+  seed - (String)the seed to initialize the key with"
+  [seed]
+  (let [keygen (KeyGenerator/getInstance "AES")
+        sr     (SecureRandom/getInstance "SHA1PRNG")]
+    (.setSeed ^SecureRandom sr (.getBytes seed "UTF-8"))
```
Comment (Scriptor) [inline src/hal/core.clj]: Should enable warn-on-reflection and see where annotations can be placed.

File: src/hal/core.clj (hunk 3 above)
Comment (bubskee) [inline src/hal/core.clj]: 4
... you nosy son of a gun

Comment (bubskee) [inline src/hal/core.clj]: (initialization vector)

Comment (mgaare) [inline src/hal/core.clj]: it's just random bytes right? the fact that it's going to be used as an initialization vector, whatever that is, doesn't really matter

Comment (bubskee) [inline src/hal/core.clj]: yea. and the only reason it isn't private is so it can have a starring role in a couple unit tests.

File: project.clj [hunk 5]
```diff
@@ -0,0 +1,13 @@
+(defproject hal "0.1.0-SNAPSHOT"
+  :description "FIXME: write description"
+  :url "http://example.com/FIXME"
+  :license {:name "Eclipse Public License"
```
Comment (willspecht) [inline project.clj]: I thought we were removing this?

Comment (bubskee) [inline project.clj]: good catch!

**Pull request merged**